.github/workflows/deploy.yml

```diff
@@ -0,0 +1,204 @@
+  DRY_RUN: false # ?
+  STEP_SCRIPTS: ${{ github.workspace }}/.github/steps/deploy
+  # target_cluster: dev
+
+jobs:
+  # The `configure` job reconciles the target version (if it wasn't explicitly provided),
+  # creates a slack notification for the deployment (except for developer instances),
+  # and updates the slack notification with additional deployment context information.
+  configure:
+    env:
+      target_cluster: ${{ github.event.inputs.target_cluster }}
+      rfc: ${{ github.event.inputs.rfc }}
+    outputs:
+      target-version: ${{ steps.reconcile-version.outputs.target-version }}
+      slack-notification-id: ${{ steps.slack.outputs.canvas-id }} #?
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # We require a deployment record for the prod endpoint to make it
+      # harder to accidentally deploy to prod; unfortunately we cannot
+      # currently validate this record, because endpoints usually require
+      # authentication.
+      - name: Verify prod deployment record
+        if:  github.event.inputs.target_cluster == 'prod'
+        run: |
+          if [[ -z "${{ env.rfc }}" ]]
+          then
+            echo "Deployment to prod requires a link to an rfc"
+            exit 1
+          fi
+      # If the entity who created this deployment did not provide a version,
+      # we will derive the version from the target stage's "previous" environment.
+      # This means that deployments to eval will promote from dev, and deployments
+      # to prod will promote from eval. It's OK if the same version is re-deployed,
+      # it will have the same net effect of a `kubectl rollout restart`.
+      # Output: `target-version`: The reconciled version to deploy.
+      - name: Reconcile target deployment version
+        id: reconcile-version
+        run: |
+          set -x
+          if [[ -z "${target_version}" ]]
+          then
+             case "${target_cluster}" in
+                 prod)
+                   source_stage=eval
+                   ;;
+                 dev)
+                   source_stage=poetry
+                   ;;
+                 *)
+                   source_stage=dev
+                   ;;
+             esac
+             target_version=$(./scripts/get-deployed-version.sh -s ${target_cluster})
+          fi
+          if [[ "${source_stage}" == "poetry" ]]
+          then
+            # Get the target version from the poetry configuration
+            source ./scripts/globals.sh
+            target_version=$(get_poetry_version)
+          else
+            target_version=$(./scripts/get-deployed-version.sh -s ${source_stage})
+          fi
+          echo "::set-output name=target-version::${target_version}"
+      # For shared instances (dev, eval, prod) we create a slack notification
+      # so that others can be aware of the change.
+      # TODO: Eval and prod deployment notifications should go to #iam-uwnetid
```

Comment:
Is there an issue associated with this TODO?

Reply:
Thanks Mike! There is now ;) https://github.com/UWIT-IAM/uw-husky-directory/issues/152